src/main/kotlin/skill/TaoQu.kt

```diff
@@ -123,13 +125,29 @@ class TaoQu : MainPhaseSkill() {
             }
             if (r is RobotPlayer) {
                 GameExecutor.post(g, {
-                    val playerAndCard = g.players.flatMap {
-                        if (!it!!.alive || !it.isEnemy(r)) emptyList()
-                        else it.messageCards.mapNotNull { card ->
-                            if (!card.colors.any { c -> c in colors }) null
-                            else PlayerAndCard(it, card)
+                    val color = listOf(Red, Blue, Black).filter {
+                        r.cards.count(it) >= 2
+                    } // 手牌中出现两次的颜色
+                    val movecards = ArrayList<Card>()
+                    val playerAndCard: PlayerAndCard?
+                    var value = Int.MIN_VALUE
+                    for (p in g.players) {
+                        p!!.alive || continue // 死亡玩家跳过
+                        for (movecard in p.messageCards) {
+                            movecard.colors.any { color.contains(it) } || continue // 遍历到没有任意两张手牌含有相同的颜色跳过
+                            val v = r.calculateRemoveCardValue(r, p, movecard)
+                            if (v > value) {
+                                value = v
+                                movecards.clear()
+                                movecards.add(movecard)
+                            } else if (v == value) {
+                                movecards.add(movecard)
+                            }
                         }
-                    }.minBy { RobotPlayer.cardOrder[it.card.type]!! }
+                    }
+                    val mostcolor = movecards.flatMap { it.colors }.groupingBy { it }.eachCount().maxBy { it.value }.key
+                    val movecard = movecards.sortCards(mostcolor, false)[0]
```

Comment:
这里没加好注释，`sortCards`的第一个参数传发动技能的人的身份颜色。顺便加个注释吧。（同理还有下面的那个`bestCard`和`bestCardOrNull`方法）
```suggestion
                    val movecard = movecards.sortCards(r.identity)[0]
```